src/libopenrave/robot.cpp

```diff
@@ -245,6 +260,19 @@ void RobotBase::RobotStateSaver::_RestoreRobot(boost::shared_ptr<RobotBase> prob
             }
         }
     }
+    if( _options & Save_ManipulatorsToolTransform ) {
+        if( probot == _probot ) {
+            std::vector<RobotBase::ManipulatorPtr> vmanips = probot->GetManipulators();
+            for(int imanip = 0; imanip < vmanips.size(); ++imanip){
+                RobotBase::ManipulatorPtr pmanip = vmanips[imanip];
+                if( !!pmanip ){
+                    pmanip->SetLocalToolTransform(_vtManipsLocalTool[imanip]);
```

Comment:
what if robot manipulators changed and they are not the same number? you have to either do better matching, or throw exception, right now this will segfault.